pkgs/applications/video/davinci-resolve/default.nix

```
@@ -1,179 +1,23 @@
-          -H 'Accept-Encoding: gzip, deflate, br' \
-          -H 'Accept-Language: en-US,en;q=0.9' \
-          -H 'Authority: www.blackmagicdesign.com' \
-          -H 'Cookie: _ga=GA1.2.1849503966.1518103294; _gid=GA1.2.953840595.1518103294' \
-          --data-ascii "$REQJSON" \
-          --compressed \
-          "$SITEURL")
-
-        curl \
-          --retry 3 --retry-delay 3 \
-          -H "Host: sw.blackmagicdesign.com" \
-          -H "Upgrade-Insecure-Requests: 1" \
-          -H "$USERAGENT" \
-          -H "Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8" \
-          -H "Accept-Language: en-US,en;q=0.9" \
-          --compressed \
-          "$RESOLVEURL" \
-          > $out
-      '';
-
-      # The unpack phase won't generate a directory
-      setSourceRoot = ''
-        sourceRoot=$PWD
-      '';
-
-      installPhase = ''
-        runHook preInstall
-
-        export HOME=$PWD/home
-        mkdir -p $HOME
-
-        mkdir -p $out
-        appimage-run ./DaVinci_Resolve_${version}_Linux.run -i -y -n -C $out
-
-        mkdir -p $out/{configs,DolbyVision,easyDCP,Fairlight,GPUCache,logs,Media,"Resolve Disk Database",.crashreport,.license,.LUT}
-        runHook postInstall
-      '';
-
-      dontStrip = true;
-
-      postFixup = ''
-        for program in $out/bin/*; do
-          isELF "$program" || continue
-          addOpenGLRunpath "$program"
-        done
-
-        for program in $out/libs/*; do
-          isELF "$program" || continue
-          if [[ "$program" != *"libcudnn_cnn_infer"* ]];then
-            echo $program
-            addOpenGLRunpath "$program"
-          fi
-        done
-      '';
-    }
-  );
 in
-buildFHSUserEnv {
-  name = "davinci-resolve";
-  targetPkgs = pkgs: with pkgs; [
-    librsvg
-    libGLU
-    libGL
-    xorg.libICE
-    xorg.libSM
-    xorg.libXxf86vm
-    xorg.libxcb
-    udev
-    opencl-headers
-    alsa-lib
-    xorg.libX11
-    xorg.libXext
-    expat
-    zlib
-    libuuid
-    bzip2
-    libtool
-    ocl-icd
-    glib
-    libarchive
-    xdg-utils # xdg-open needed to open URLs
-    python
-    # currently they want python 3.6 which is EOL
-    #python3
-  ];
+rec {
 
-  runScript = "${bash}/bin/bash ${
-    writeText "davinci-wrapper"
-    ''
-    export QT_XKB_CONFIG_ROOT="${xkeyboard_config}/share/X11/xkb"
-    export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:${davinci}/libs
-    ${davinci}/bin/resolve
-    ''
-  }";
+  stable = generic {
+    version = "17.4.3";
+    pname = "davinci-resolve";
+  };
 
-  meta = with lib; {
-    description = "Professional Video Editing, Color, Effects and Audio Post";
-    homepage = "https://www.blackmagicdesign.com/products/davinciresolve/";
-    license = licenses.unfree;
-    maintainers = with maintainers; [ jshcmpbll ];
-    platforms = platforms.linux;
+  beta = generic {
+    version = "17.4.6";
+    pname = "davinci-resolve";
```

please use attribute name
```suggestion
    pname = "davinci-resolve-beta";
```